user-service/src/main/java/sjchat/users/UserAuthentication.java

```diff
@@ -0,0 +1,126 @@
+package sjchat.users;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import java.lang.Exception;
+import java.util.Date;
+import sjchat.daos.UserDao;
+import sjchat.daos.UserDaoImpl;
+import sjchat.entities.UserEntity;
+import sjchat.exceptions.NoEntityExistsException;
+
+import sjchat.users.exceptions.AuthenticationException;
+import sjchat.users.tokens.*;
+
+public class UserAuthentication {
+  private TokenConfig.Configurations config;
+  private TokenAuth auth;
+  private TokenBuilder builder;
+  private final Date nulldate;
+  private final UserDao dao;
+
+  public void updateConfiguration() {
+    TokenConfig.refreshConfiguration();
+    this.config = TokenConfig.get();
+    this.auth = new TokenAuth(this.config.issuer, this.config.secret);
+    this.builder = new TokenBuilder(this.config.issuer, this.config.secret);
+  }
+
+  private UserAuthentication() {
+    this.config = TokenConfig.get();
+    this.auth = new TokenAuth(this.config.issuer, this.config.secret);
+    this.builder = new TokenBuilder(this.config.issuer, this.config.secret);
+    this.nulldate = new Date(0);
+    this.dao = new UserDaoImpl();
+  }
+
+
+  public static UserAuthentication getInstance() {
+    return UserAuthenticationHolder.INSTANCE;
+  }
+
+
+  private static class UserAuthenticationHolder {
+    private static final UserAuthentication INSTANCE = new UserAuthentication();
+  }
+
+  public UserAuthentication checkUserExists(String username, String password) throws AuthenticationException {
+    UserEntity user = dao.findByUsername(username);
+
+    if (user == null) {
+      throw new AuthenticationException("Bad username");
+    }
+
+    if (user.getPassword() != password) {
+      throw new AuthenticationException("Bad password");
+    }
+
+    return this;
+  }
+
+  public UserAuthentication checkUsernameExists(String username) throws AuthenticationException {
```

Comment:
Indeed & The UserUpdate static class should probably be to.. Unless i rename the entire class to something more fitting? I kind of like the flow of having that functionality in there. It made the code in UserAuth look more clean in my opinion. But i can change it.